l10n_it_rea/models/res_company.py

```diff
@@ -0,0 +1,55 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 Lorenzo Battistini
+
+from odoo import fields, models, api, _
+from odoo.tools.misc import formatLang
+
+
+class Company(models.Model):
+    _inherit = 'res.company'
+
+    rea_office = fields.Many2one(
+        'res.country.state', string='Office Province',
+        related='partner_id.rea_office')
+    rea_code = fields.Char('REA Code', size=20, related='partner_id.rea_code')
+    rea_capital = fields.Float('Capital', related='partner_id.rea_capital')
+    rea_member_type = fields.Selection(
+        [('SU', 'Unique Member'),
+         ('SM', 'Multiple Members')], 'Member Type',
+        related='partner_id.rea_member_type')
+    rea_liquidation_state = fields.Selection(
+        [('LS', 'In liquidation'),
+         ('LN', 'Not in liquidation')], 'Liquidation State',
+        related='partner_id.rea_liquidation_state')
+    company_registry = fields.Char(
+        "Company Registry", compute="_compute_company_registry", store=True)
+
+    @api.depends(
+        "partner_id.rea_office", "partner_id.rea_code",
+        "partner_id.rea_capital", "partner_id.rea_member_type",
+        "partner_id.rea_liquidation_state"
+    )
+    @api.multi
+    def _compute_company_registry(self):
+        for company in self:
+            rea_member_type = ''
+            if company.rea_member_type:
+                rea_member_type = dict(
+                    self.env['res.partner']._fields[
+                        'rea_member_type'
+                    ]._description_selection(self.env)
+                )[company.rea_member_type]
+            rea_liquidation_state = ''
+            if company.rea_liquidation_state:
+                rea_liquidation_state = dict(
+                    self.env['res.partner']._fields[
+                        'rea_liquidation_state'
+                    ]._description_selection(self.env)
+                )[company.rea_liquidation_state]
+            company_registry = _("%s %s Share Cap. %s%s %s %s") % (
```

Comment:
Check `currency.position` to see if the symbol goes before or after the amount, or try to take advantage of a `Monetary` field

Reply:
@SimoRubi do you know in this case how could I take advantage of a Monetary field?

Reply:
That would be more nicely displayed in the company form, actually I don't know how that could be useful in string substitution